### ISSUE TITLE
Getting rid of ffmpeg

### ISSUE THREAD

**Issue by Bnyro** [libre-tube/LibreTube#395]
### Describe your suggested feature

ffmpeg increases the app size by 50% and isn't the most reliable library. I think we should try to get rid of it. I would really appreciate if there's any dev around here who has an idea how to mux audio and video without using ffmpeg and could help us.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Legoks** [comment]
I think HandBrake is a better alternative.

**Comment by Legoks** [comment]
Video Transcoder for Android!

**Comment by Legoks** [comment]
HandBrake is not compatible for android

**Comment by Bnyro** [comment]
It uses ffmpeg too :)

**Comment by Pentaphon** [comment]
How does Newpipe do it?

**Comment by Nizami20052022** [comment]
The NewPipeExtractor is a Java framework for scraping video platform websites in a way that they can be accessed like a normal API. The extractor is the core of the popular YouTube and streaming app [NewPipe](https://newpipe.schabi.org/) for Android.

**Comment by Pentaphon** [comment]
> The NewPipeExtractor is a Java framework for scraping video platform websites in a way that they can be accessed like a normal API. The extractor is the core of the popular YouTube and streaming app [NewPipe](https://newpipe.schabi.org/) for Android.

Well I doubt we want to use that or a fork of that so I guess we may have to stick with ffmpeg unless somebody wants to create a from-scratch Libretube equivalent of NewpipeExtractor.

**Comment by FireMasterK** [comment]
> Well I doubt we want to use that or a fork of that so I guess we may have to stick with ffmpeg unless somebody wants to create a from-scratch Libretube equivalent of NewpipeExtractor. 

That won't help you since NewPipeExtractor doesn't help with muxing. Libretube uses Piped which uses NewPipeExtractor.

What NewPipe does is magic similar in https://github.com/TeamNewPipe/NewPipe/blob/0039312a64e7271fec2e0a295db1d6979e83cf3f/app/src/main/java/org/schabi/newpipe/streams/Mp4FromDashWriter.java

**Comment by Bnyro** [comment]
@FireMasterK Yeah, that's actually what I found too but it's kinda hard to understand how it works ...
